src/main/ts/components/Editor.tsx

```diff
@@ -85,6 +85,9 @@ export class Editor extends React.Component<IAllProps> {
 
   public componentWillUnmount() {
     if (getTinymce() !== null) {
+      if (this.editor) {
+        this.editor.off('SaveContent');
```

Comment:
Hmm... this will prevent the content being saved when the editor is being shutdown/removed. As it does a save call to make sure it's up to date. The `SaveContent` event does pass through a `is_removing` flag in the event, so could you ignore the event if that flag is set?

If we do go ahead with this, I'd be very worried it could break some users integrations and as such I'd suggest this be put behind an opt in flag. Maybe something like `saveOnDestroy`?

Reply:
It's also worth mentioning, that the `SaveContent` event possibly doesn't hold a lot of meaning for the TinyMCE integrations, as it's fired when saving the content back to the original target element which is created by the integration. So maybe instead it shouldn't exist in the integrations.

Additionally looking at the attached issue, it seems that the `SaveContent` event is trying to be used when the "Save" button is pressed? If so, the `save_onsavecallback` init setting should be used for that purpose instead. We could look at adding a new event for that as well, but that'd be best requested via the TinyMCE repo instead.

Reply:
If people are using a react integration then I'd expect they'd get the editor content in the appropriate stages of the react lifecycle so I don't expect there would be a problem.

Reply:
I mean the very fact that it's got a `componentWillUnmount` suggests that it has already been removed by some state change.

Reply:
So therefore I would have expected that state change to have stored the value if it was important to do so.

Reply:
> I mean the very fact that it's got a `componentWillUnmount` suggests that it has already been removed by some state change.

This is incorrect, as the name states it's about to unmount, it has not yet removed the component from view. This is also where any state would normally be saved if required.

Reply:
Looking at this more, I wonder if we should instead be unbinding all events before destroying TinyMCE? That however would be a pretty breaking change, which could break things for a number of customers.

Reply:
Why not a list of events to unbind as a wrapper option?

Reply:
> Why not a list of events to unbind as a wrapper option?

Yeah, that'd work pretty nicely as well :+1:

Reply:
> should instead be unbinding all events before destroying TinyMCE

That sounds sensible to me.